picard/script/__init__.py

```diff
@@ -101,3 +114,257 @@ def enabled_tagger_scripts_texts():
     if not config.setting["enable_tagger_scripts"]:
         return []
     return [(s_name, s_text) for _s_pos, s_name, s_enabled, s_text in config.setting["list_of_scripts"] if s_enabled and s_text]
+
+
+@unique
+class PicardScriptType(IntEnum):
+    """Picard Script object types
+    """
+    BASE = 0
+    TAGGER = 1
+    FILENAMING = 2
+
+
+class PicardScript():
+    """Base class for Picard script objects.
+    """
+    # Base class developed to support future tagging script class as possible replacement for currently used tuples in config.setting["list_of_scripts"].
+
+    TYPE = PicardScriptType.BASE
+    JSON_OUTPUT = {'title', 'script'}
+
+    # Don't automatically trigger changing the `script_last_updated` property when updating these properties.
+    _last_updated_ignore_list = {'last_updated', 'readonly', 'deletable', 'id'}
+
+    def __init__(self, script='', title='', id=None, last_updated=None):
+        """Base class for Picard script objects
+
+        Args:
+            script (str): Text of the script.
+            title (str): Title of the script.
+            id (str): ID code for the script. Defaults to a system generated uuid.
+            last_updated (str): The UTC date and time when the script was last updated. Defaults to current date/time.
+        """
+        self.title = title if title else DEFAULT_SCRIPT_NAME
+        self.script = script
+        if id is None:
+            self._set_new_id()
+        else:
+            self.id = id
+        if last_updated is None:
+            self.update_last_updated()
+        else:
+            self.last_updated = last_updated
+
+    def _set_new_id(self):
+        """Sets the ID of the script to a new system generated uuid.
+        """
+        self.id = str(uuid.uuid4())
+
+    def get_value(self, setting):
+        """A safe way of getting the value of the specified property setting, because
+        it handles missing properties by returning None rather than raising an exception.
```

Comment:
What do you think about maybe making this dict like by implementing `__getitem__`? Or if we want to keep it as attributes implement `__getattr__`?

Reply:
Much nicer!  Done.